streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/FluxComponentFactory.java

```diff
@@ -33,7 +33,7 @@
     FluxComponent getFluxComponent(StreamlineComponent streamlineComponent) {
         ProxyUtil<FluxComponent> proxyUtil = new ProxyUtil<>(FluxComponent.class);
         try {
-            FluxComponent fluxComponent = proxyUtil.loadClassFromJar(extraJarsLocation.toAbsolutePath().toString(), streamlineComponent.getTransformationClass());
+            FluxComponent fluxComponent = proxyUtil.loadClassFromLibsDirectory(extraJarsLocation, streamlineComponent.getTransformationClass());
```

Comment:
Is the user expected to build and place the jar into the "/tmp/storm-artifacts/jars" directory ?

The extrasJarsLocation is already a dir. I think UrlClassLoader already handles directories. 

https://docs.oracle.com/javase/7/docs/api/java/net/URLClassLoader.html
"This class loader is used to load classes and resources from a search path of URLs referring to both JAR files and directories. Any URL that ends with a '/' is assumed to refer to a directory. Otherwise, the URL is assumed to refer to a JAR file which will be opened as needed."

Reply:
The user is not expected to do that. The curl call used to register a component bundle accepts a jar and it stores it under file storage. While deployment we keep track if there is bundle that is non built in. If so, we download jar from storage and put it under /tmp/storm-artifacts/jars. 

As far as UrClassLoader is concerned, i read that javadoc as well and tried putting a / at the end of extraJarsLocation to see why existing method did not work. Digging more, I found that when URLClassLoader interprets it as a directory, it actually tries to search that directory for a class using the package hierarchy. So for a URLClassLoader to load a class a.b.c.Someclass, it expects the entire directory structure corresponding to the package hierarchy and finally a .class file for that class. Hence, I had to add the other method where it would load all the jar files in the directory. This in java world is usually referred to as a "libs" directory where you put all your jars. Which is why I had used the word "libs" instead of "lib". However, I don't have a opinion on that and have renamed it as per your suggestion